shared/bundle_analysis/comparison.py

```diff
@@ -176,6 +188,68 @@ def _match_assets(
         return matches
 
 
+class BundleRoutesComparison:
+    """
+    Compares all routes of two bundle route reports for a given bundle
+    """
+
+    def __init__(
+        self,
+        base_report: BundleRouteReport,
+        head_report: BundleRouteReport,
+    ):
+        self.base_report = base_report
+        self.head_report = head_report
+
+    @sentry_sdk.trace
+    def size_changes(self) -> List[RouteChange]:
+        """
+        Returns a list of changes for each unique route that exists between the base and head.
+        If a route exists on base but not head that is considered "removed" and -100% percentage delta
+        If a route exists on head but not base that is considered "added" and +100% percentage delta
+        Otherwise it is considered "changed" and percentage delta = (diff_size / base_size) * 100
+        """
+        base_sizes = self.base_report.get_size()
+        head_sizes = self.head_report.get_size()
+
+        all_routes, results = base_sizes.keys() | head_sizes.keys(), []
+        for route_name in all_routes:
+            # Added new route
+            if route_name not in base_sizes:
```

Comment:
Do we need to add the case where base_sizes[route_name] is 0 to avoid a potential divide by 0 issue at line 240? Can 0 even happen - maybe with rounding?